File: qualification/a01/fixture/a01-code-qual-001/live-candidate/python/canonical.py
from __future__ import annotations

import hashlib


def canonical(request_id: str, payload: str) -> str:
    # Intentionally defective: does not normalize incidental whitespace.
    return f"{request_id}:{payload}"


def sha256(request_id: str, payload: str) -> str:
    return hashlib.sha256(canonical(request_id, payload).encode("utf-8")).hexdigest()

File: qualification/a01/fixture/a01-code-qual-001/live-candidate/javascript/canonical.js
import { createHash } from 'node:crypto';

export function canonical(requestId, payload) {
  // Intentionally defective: differs from the required cross-language contract.
  return `${requestId}:${payload}`;
}

export function sha256(requestId, payload) {
  return createHash('sha256').update(canonical(requestId, payload), 'utf8').digest('hex');
}
